From 515c12fa27983fd8852ef67067112823ef28dc0d Mon Sep 17 00:00:00 2001
From: BambooIV <ningdisheng@gmail.com>
Date: Tue, 15 Mar 2016 11:48:31 +0800
Subject: [PATCH 04/36] nx503a: msm: krait-regulator: handle efuse differences

cherry-pick from 5c50e2f56bf8218ef68cffeb583cc7d8d8bb2ce4

Scaling factor version is valid for
- 8974 based devices when the version bits read 1

Moreover the three bit scaling factor data for
- 8974 based devices starts at bit 16

Update the code to handle these differences.
---
 arch/arm/boot/dts/NX503A/msm8974-regulator.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boot/dts/NX503A/msm8974-regulator.dtsi b/arch/arm/boot/dts/NX503A/msm8974-regulator.dtsi
index 962c956..55da718 100644
--- a/arch/arm/boot/dts/NX503A/msm8974-regulator.dtsi
+++ b/arch/arm/boot/dts/NX503A/msm8974-regulator.dtsi
@@ -465,6 +465,8 @@
 		ranges;
 		qcom,pfm-threshold = <76>;
 		qcom,use-phase-scaling-factor;
+		qcom,phase-scaling-factor-bits-pos = <16>;
+		qcom,valid-scaling-factor-versions = <0 1 0 0>;
 
 		krait0_vreg: regulator@f9088000 {
 			compatible = "qcom,krait-regulator";
-- 
1.9.1

